mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -285,9 +286,9 @@ fun JSONObject.collectWidgets(parent: Widget?): List<Widget> {
     val type = getString("type").toWidgetType()
     val icon = optStringOrNull("icon")
     val (minValue, maxValue, step) = Widget.sanitizeMinMaxStep(
-        optDouble("minValue", 0.0).toFloat(),
-        optDouble("maxValue", 100.0).toFloat(),
-        optDouble("step", 1.0).toFloat()
+        optFloatOrNull("minValue") ?: item?.minimum ?: 0f,
+        optFloatOrNull("maxValue") ?: item?.maximum ?: 100f,
+        optFloatOrNull("step") ?: item?.step ?: 1f
```

Comment:
Need to update `updateFromEvent` with this logic, or convert min/max/step to a getter function.

Reply:
Done by adding getters.